### PR TITLE
Add git .mailmap file

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

No JIRA, just a little fun `git` hack. Makes the names in the log nice. I only used information you could find in the log itself - the classic pattern is that someone contributes for a while before setting up their `git` configuration to have good metadata. Or in this case, also there was a tool that doesn't preserve good metadata.

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @davorbonaci perhaps

**Comment by davorbonaci** [comment]
+1 that it makes mapping much, much nicer.
+1 that is used by several other Apache projects.

-1 that is makes email addresses easy to collect. (Sure, they are already publicly available, but there's a meaningful extra effort to scrape them from git, which makes it not worth the effort for others.)
-1 on the administrative burden to keep it useful.

The benefit is limited to command-line usage of git; GitHub has a separate mapping database.

Also, other cases seem to look slightly different:
* https://github.com/apache/climate/blob/master/.mailmap
* https://github.com/apache/incubator-edgent/blob/master/.mailmap
* https://github.com/apache/couchdb/blob/master/.mailmap

Overall, I think we can go either way. Personally, I think cons outweigh the benefits, so I'm slightly opposed, but won't object if others want to pursue this. If you'd like to proceed, please wait for at least a few named individuals are fine being listed here.

**Comment by kennknowles** [comment]
The benefit is actually not just command-line `git` itself but other mature tools that use the git history. GitHub is a reasonable exception, since they have their own protocol for email claiming.

Anyhow I look at the history far more often through `git` than through GitHub, especially in interesting cases. So, personally, it is already nice even if that is the only benefit. Probably I am not the only one.

I considered your cons before putting this together, but have these thoughts:

 - Actually every email address here is already present in the git log. Anyone writing a scraper will go there first, since it is more common. I deliberately have not added any information. If anything, it will actually make some scrapers get fewer emails :-). I will seek approvals from people listed though. 
 - The administrative burden is a good thing to think of, but I propose to not centrally administer the list, thus making the burden zero. I am offering this initial list as a volunteer effort on my part. After that, it can just be as folks desire. You'll also notice that a lot of the bad names are just from tool-generated history, so the burden is much lower going forwards. (more contributors, but fewer needed in this file)
 - As for the format, there are a few different formats and you can choose them on a line-by-line basis. I chose one that seemed very simple. I've checked it out and didn't see obvious issues, but it is my first attempt at a `.mailmap`.

So in summary in my opinion we can easily choose for the cons to measure zero, but also the pros are not very important. Just some nice polish.

**Comment by davorbonaci** [comment]
Ack on other comments; one comment however.

> Anyone writing a scraper will go there first, since it is more common.

Indeed, as we both commented, this is already public information in "git log". However, I wanted to point out web crawling way of scrapping things. Before this PR, web crawler probably wouldn't get it; after it, it will. This is the most common. However, an argument against this is: ~sooner or later it will leak somewhere else, not worth trying to prevent it.

Again, I'm (very) slightly against, but if you like it, LGTM.

**Comment by kennknowles** [comment]
Since there's two mild votes against, I'll drop it. If I need high fidelity mapping I can just cherry pick it back in and touch it up. We won't benefit from community maintenance but that isn't such a huge deal.

**Comment by kennknowles** [comment]
Sorry, misread. Just one mild vote against :-) 

But that's fine.
